### ISSUE TITLE
Example of adding additional branches

### ISSUE THREAD

**Issue by masonproffitt** [iris-hep/adl-benchmarks-index#6]
From Andrea Rizzi:

>In another thread we recently had a discussion on the flexibility to
"add additional branches" (originating from some parent data file) and
it would also be nice to add such kind of examples.

**Comment by masonproffitt** [comment]
From a [conversation](https://github.com/iris-hep/adl-benchmarks-index/issues/16#issuecomment-808512672) in #16: We can assume that the underlying system can handle added columns using exactly the same interface as the existing ones, but we should add a note about this assumption in case a system does not have this feature.

**Comment by ingomueller-net** [comment]
I agree that #16 is a duplicate of this one. I shall add that the system, in particular, needs to be able to produce data in a format that can be used as *input* of other queries. In contrast, all of the current queries produce histograms, which may be plotted and which it would not make sense to query again. (Indeed, in [Presto](https://github.com/RumbleDB/iris-hep-benchmark-presto) it is quite inconvenient to *produce* new arrays, which I why I had opened #16.)
